cmake/std/trilinosprhelpers/setenvironment/test/test_SetEnvironment.py

```diff
@@ -0,0 +1,520 @@
+#!/usr/bin/env python
+# -*- coding: utf-8; mode: python; py-indent-offset: 4; py-continuation-offset: 4 -*-
+"""
+"""
+from __future__ import print_function
+import sys
+sys.dont_write_bytecode = True
+
+import os
+sys.path.insert(1, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from pprint import pprint
+
+import unittest
+from unittest import TestCase
+
+# Coverage will always miss one of these depending on the system
+# and what is available.
+try:                                               # pragma: no cover
+    import unittest.mock as mock                   # pragma: no cover
+except:                                            # pragma: no cover
+    import mock                                    # pragma: no cover
+
+from mock import Mock
+from mock import MagicMock
+from mock import patch
+
+try:
+    from cStringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+from trilinosprhelpers.setenvironment import SetEnvironment
+
+
+#===============================================================================
+#
+# Mock Helpers
+#
+#===============================================================================
+
+
+def mock_module_pass(*args):
+    """
+    Mock the module() command that 'passes', returning a 0.
+    """
+    print("mock> module({}) ==> 0".format(args))
+    return 0
+
+
+def mock_module_fail(*args):
+    """
+    Mock the module() command that 'fails', returning a 1.
+    """
+    print("mock> module({}) ==> 1".format(args))
+    return 1
+
+
+#===============================================================================
+#
+# Tests
+#
+#===============================================================================
+
+class SetEnvironmentTest(TestCase):
+    """
+    Main test driver for the SetEnvironment class
+    """
+    def setUp(self):
+        print("")
+        self.maxDiff = None
+
+        self._filename = self.find_config_ini(filename="setenvironment_config.ini")
+
+        self._test_profile_001_truth = {
+            'module-list': {
+                'sems-gcc': True,
+                'sems-boost': True,
+                'sems-cmake': True,
+                'sems-python': True
+            },
+            'module-op': [
+                ['purge', ''],
+                ['use', '/projects/sems/modulefiles/projects'],
+                ['load', 'sems-env'],
+                ['load', 'sems-gcc/7.3.0'],
+                ['load', 'sems-boost/1.63.0/base'],
+                ['load', 'sems-cmake/3.10.3'],
+                ['load', 'sems-python/3.5.2']
+            ],
+            'setenv': [   {'key': 'OMP_NUM_THREADS', 'value': '2'},
+                          {'key': 'CC', 'value': 'gcc'},
+                          {'key': 'CXX', 'value': 'g++'},
+                          {'key': 'FC', 'value': 'gfortran'},
+                          {'key': 'TEST_ENVVAR_001', 'value': 'TEST_ENVVAR_001_VALUE'}
```

Comment:
I'm kind of surprised the `setenv` thing builds up a list of dicts instead of just a dict.

Reply:
a list of dicts that have identical keys.

Note that each dict in this has the keys "key" and "value" ... when I iterate over this, it looks for the "key" and "Value" properties.

Reply:
Oh, gotcha.